File: Q2.py
def fa(n,prod):
    try:
        k=int(n)
        while k<0:
            print("The input is negative")
            k = int(input("Please enter a non-negative integer "))
        for val in range(1,k+1):
            prod = prod*val
        print("The factorial is",prod)
    except ValueError:
        c=input("Error: The number you have entered is not an integer. Try again: ")
        fa(c,1)
b = input("Enter the last number of the sum: ")
fa(b,1)

File: Q2.cpp
#include<iostream>
using namespace std;
bool ion(float n){
    while(n>=0){
        if(n==0)return true;
        n--;
    }
    return false;
}
int main(){
    float c=1,n;
    cout<<"Enter the number for which you want to find the factorial :";
    cin>>n;
    while(ion(n)==false){
        cout<<"Please enter a non-negative integer "<<endl;
        cin>>n;
    }
    for(int i=n;i>0;i--)
    {
        c*=i;
    }
    cout<<"The factorial is "<<c;
    
    

}